Src/BootCamp.Chapter/Lesson3.cs

```diff
@@ -0,0 +1,126 @@
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static float IsNumber(string input)
+        {
+            var isEmpty = String.IsNullOrEmpty(input);
+           
+            if (isEmpty) return 0f;
+            
+            if ((input) == "0") return 0f;
+
+            var isNumber = float.TryParse(input, out var result);
+            if (!isNumber) 
+            {
+                Console.Write(@"""" + input + @""" " + "is not a valid number.");
+
+                return -1f;
+            }
+            
+            return result;
+        }
+        
+        public static string GetString(string message)
+        {
+            Console.WriteLine(message);
+            
+            var input = Console.ReadLine();
+
+            bool isEmpty = String.IsNullOrEmpty(input);
+            if (isEmpty)
+            {
+                Console.Write("Name cannot be empty.");
+
+                return "-";
+            }
+
+            return input;
+        }
+       
+        public static int GetInt(string message)
+        {
+            Console.WriteLine(message);
+            
+            string int1 = Console.ReadLine();
+            
+            var isInt = IsNumber(int1);
+            
+            int result = Convert.ToInt32(isInt);
+            
+            return (result);
+        }
+       
+        public static float GetFloat(string message)
+        {
+            Console.WriteLine(message);
+            
+            string float1 = Console.ReadLine();
+            
+            var isFloat = IsNumber(float1);
+           
+            return isFloat;
+        }
+
+        public static float Bmi(float weight, float height)
+        {
+            string heightError = ("Height cannot be equal or less than zero, but was " + height);
+            string weightError = ("Weight cannot be equal or less than zero, but was " + weight);
+            string failedCalc = ("Failed calculating BMI. Reason:");
+
+            bool isW0OrLess = ((weight) <= 0);
+            bool isH0OrLess = ((height) <= 0);
+
+            if (isH0OrLess && isW0OrLess)
+            {
+                Console.WriteLine(failedCalc);
+                Console.WriteLine(weightError + '.' + Environment.NewLine + "Height cannot be less than zero, but was " + height + '.');
+
+                return -1f;
+            }
+
+            if (isW0OrLess)
+            {
+                Console.WriteLine(failedCalc);
+                Console.WriteLine(weightError + '.');
+                return -1f;
+            }
+
+            if (isH0OrLess)
+            {
+                Console.WriteLine(failedCalc);
+                Console.WriteLine(heightError + '.');
+                return - 1f;
+            }
+
+            float bmi = (weight / height / height);
+            
+            return bmi;
+        }
+       
+        public static void Demo()
+        {
+            static void GatherAndPrint(string userNo)
```

Comment:
This should be moved outside of function to class scope. Nested function are harder to read, because they polute the function they are at by rubbing all the implementation details we don't need to care about.

Reply:
I'm not sure if I've done what you've meant

Reply:
It's still there. ``static void GatherAndPrint(string userNo)`` should be outside of Demo method, inside Lesson3 class directly.

Reply:
I'm confused. The demo function is just this:
public static void Demo()
        {
            Process2PersonInfo();
        }

Should the Process2PersonInfo() be called by some other function, and demo left empty? But if it's empty how is the main function going to activate the program?